doc/fast-reboot/Fast-reboot_Flow_Improvements_HLD.md

```diff
@@ -0,0 +1,158 @@
+
+### NOTICE
+
+'warmRestoreValidation' might fail the operation just like in fastfast-reboot case, if the way orchagent process an event from the DB is handled differently with the new software version the task will fail to execute and fast-reboot will fail along with it.
+This is solvable by the db migrator.
+
+## 4.2 Syncd Point Of View - INIT/APPLY view framework
+
+Syncd starts with the fast-reboot flag, trigger the ASIC reset when create_switch is requested from orchagent.
+In addition, on this case temp view flag will set to false since it is not required, no comparison logic needed since current view is empty.
+Basically INIT and APPLY view requests from orchagent are ignored by syncd, but bound the process from start to end.
+During reconsilations process of orchagent, syncd will recieve all tasks to restore the previous state.
+All other network applications will do the same as we do today for warm-reboot.
+
+![Syncd](/doc/fast-reboot/Orchagent_Syncd.svg)
+
+## 4.3 neighsyncd Point Of View
+
+Neighbors configuration is a crucial part of the L3 switch software. It is best when the neighbor configuration on the hardware is in sync with the actual switch neighbors on the network. It can't be assumed that neighbors won't change during warm restart window, while the software is restarting, the SONiC switch software has to be ready for scenarios in which during the restart window:
+ - Existing neighbors went down, e.g: VMs crashed on the server connected to ToR switch which undergoes fast-reboot.
+ - New neighbors appeared on the network, e.g: VMs created on the server connected to ToR switch which undergoes fast-reboot.
+ - MAC changes, e.g: VMs re-created or re-configured on the server connected to ToR switch which undergoes fast-reboot.
+
+![Neighbors](/doc/fast-reboot/Neighbors.svg)
+
+## 4.4 fpmsyncd Point Of View
+
+ - When BGP on a router restarts, all the BGP peers detect that the session went down and then came up. This "down/up" transition results in a "routing flap" and causes BGP route re-computation, generation of BGP routing updates, and unnecessary churn to the forwarding tables.
+ - BGP capability, termed "Graceful Restart Capability", is defined that would allow a BGP speaker to express its ability to preserve forwarding state during BGP restart.
+ - An UPDATE message with no reachable Network Layer Reachability Information (NLRI) and empty withdrawn NLRI is specified as the End-of-RIB marker that can be used by a BGP speaker to indicate to its peer the completion of the initial routing update after the session is established.
+
+![BGP](/doc/fast-reboot/BGP.svg)
+
+## 4.5 Reboot finalizer
+
+Today we have a tool used for warm-reboot to collect all reconsiliation flags from the different network applications.
+This tool can be enhanced to consider fast-reboot as well and introduce a new flag indicating the end of the process.
```

Comment:
Do you need new flag? We already have a flag to detect fast-reboot. Example:
https://github.com/Azure/sonic-buildimage/blob/bc305283417a1c305bce4035547191e1d480c1d5/files/scripts/swss.sh#L65

Reply:
This flag is a dummy flag and actually I am not sure it is even used somewhere.
It is a flag with expired timer of 180 seconds, the one referred on this HLD is in sync with all network applications going through a reconciliation process.
I am referring to the ARM_REBOOT_TABLE on State DB, what we can see by running : "show warm_restart state"